gleam.php

```diff
@@ -1,21 +1,35 @@
 <?php
 /**
  * Plugin Name: Gleam
- * Plugin URI: https://gleam.io/
+ * Plugin URI: https://gleam.io/docs/faq#wordpress
  * Description: Gleam competition plugin. Requires PHP 5.3
  * Version: 1.0
  * Author: Crowd9
  * Author URI: https://github.com/Crowd9/Gleam-Wordpress
  */
 
-add_shortcode('gleam', function($atts, $content = null) {
+
+function gleam_init() {
+    echo '<script src="//js.gleam.io/e.js" async="true"></script>';
+}
+
+function gleam_shortcode($atts, $content = null) {
     $args = shortcode_atts(array(
         'url' => null
     ), $atts);
-    
+  
     if (empty($args['url'])) {
         return;
     }
-    return sprintf('<a class="e-gleam" href="%s" rel="nofollow">%s</a><script src="//js.gleam.io/e.js" async="true"></script>',
-            htmlspecialchars($args['url']), do_shortcode($content));
-});
+  
+    return 
+        sprintf(
+            '<a class="e-gleam" href="%s" rel="nofollow">%s</a>',
+            htmlspecialchars($args['url']),
+            do_shortcode($content)
+        );
+}
+
+add_filter('wp_footer', 'gleam_init');
```

Comment:
**Question**
Will this add the gleam init to only this page's footer or all pages?

Reply:
Unfortunately it will add gleam footer on all pages. It might be possible to only call when shortcode is called, by searching current post / page if any shortcode exists

https://stackoverflow.com/questions/6235338/wordpress-loading-javascript-only-where-shortcode-appears

Reply:
I think it'd be better to only have it on pages it's needed otherwise it'll impact the pagespeed unnecessarily.